integrations_testing_framework/decorators/http_mocking_decorators.py

```diff
@@ -34,3 +88,87 @@ def interceptor(*args, **kwargs):
         return interceptor
 
     return decorator
+
+
+def _before_record_response(response, **kwargs):
+    """
+    Callback for processing response before recording to the file.
+    """
+    # Supported content-types
+    supported_content_types = {
+        'application/json': _filter_json
+    }
+    # Replace content in response body, currently works only for JSON
+    body = response['body']['string']
+    # Read content-type
+    content_type = _get_content_type(response)
+    if content_type not in supported_content_types:
+        LOGGER.warning('Unsupported content-type "%s" for response update', content_type)
+        return response
+    update_function = supported_content_types[content_type]
+    try:
+        updated_body = update_function(body, skip_keys=kwargs.get('filter_data_skip_keys'))
+    except ValueError as err:
+        raise ValueError('Failed to update response body') from err
+    response['body']['string'] = updated_body
+    return response
```

Comment:
This method will replace everything in a response, however it doesn't really maintain the datatypes. E.g. a timestamp string is replaced with an X the for an email.

I haven't used the framework but I could imagine that this reduces the utility of using it to test? Would it make sense to instead only provide specific fields that should be filtered out?

Other anonymization techniques also allow to input the specific datatype and replace the content with the same type. e.g.: https://github.com/rwnx/pynonymizer

Not saying we have to do it that way, just wondering if this is really enough.

Reply:
Updated interface

filter_resp_data -> List of keys in response that should be anonymized.
filter_resp_data_except -> List of keys that should be preserved and everything else should be anonymized.